util-hadoop/src/main/java/com/google/cloud/hadoop/util/AccessTokenProvider.java

```diff
@@ -14,41 +14,39 @@
 
 package com.google.cloud.hadoop.util;
 
+import com.google.api.services.storage.StorageRequest;
+import com.google.cloud.hadoop.gcsio.authorization.StorageAccessTokenProvider;
 import java.io.IOException;
 import org.apache.hadoop.conf.Configurable;
 
 /** A provider to provide access token, and upon access token expiration, the utility to refresh. */
-public interface AccessTokenProvider extends Configurable {
-
-  /** An access token and its expiration time. */
-  class AccessToken {
-
-    private final String token;
-    private final Long expirationTimeMilliSeconds;
-
-    public AccessToken(String token, Long expirationTimeMillis) {
-      this.token = token;
-      this.expirationTimeMilliSeconds = expirationTimeMillis;
-    }
-
-    /** @return the Access Token string. */
-    public String getToken() {
-      return token;
-    }
-
-    /** @return the Time when the token will expire, expressed in milliseconds. */
-    public Long getExpirationTimeMilliSeconds() {
-      return expirationTimeMilliSeconds;
-    }
-  }
+public interface AccessTokenProvider extends Configurable, StorageAccessTokenProvider {
 
   /** @return an access token. */
   AccessToken getAccessToken();
 
+  /**
+   * @param storageRequest a storage request holding the context.
+   * @return an access token.
+   */
+  default<RequestT extends StorageRequest<?>> AccessToken getAccessToken(RequestT storageRequest) {
```

Comment:
This makes interface GCS JSON-API specific, I think that we need to make it GCS API-agnostic.

Reply:
Introduced a new type and an adapter to make this generic.